var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -328,6 +328,10 @@ def edit(self, spec, prefix):
             # NetCDF format.
             os.environ["ESMF_NETCDF"] = "nc-config"
             os.environ["ESMF_NFCONFIG"] = "nf-config"
+            if spec["netcdf-c"].satisfies("~shared"):
+                nc_config = which("nc-config")
+                nc_flags = nc_config("--libs", output=str).strip()
+                os.environ["ESMF_NETCDF_LIBS"] = nc_flags
```

Comment:
Nitpicking: you can implement [`setup_build_environment`](https://spack-tutorial.readthedocs.io/en/latest/tutorial_advanced_packaging.html?highlight=setup_build_environment#set-environment-variables-in-your-own-package) to set the environment variable.

Reply:
Maybe I'm missing something - but this change is already in a PR. https://github.com/spack/spack/pull/35249